server/src/core/distribution/run-distribution.ts

```diff
@@ -46,11 +47,19 @@ interface CreateDistributionPlanResult {
   beneficiariesSummaries:DistributionPlanBeneficiariesSummaries;
 }
 
-export async function runDonationDistribution(db: Db, args: DonationDistributionArgs): Promise<DonationDistributionEvent[]> {
+export async function runDonationDistribution(db: Db, args: DonationDistributionArgs, filter: BeneficiaryFilter): Promise<DonationDistributionEvent[]> {
   const { periodLength, periodLimit, users } = args;
 
-  const beneficiaries = await findEligibleBeneficiaries(db, periodLimit, periodLength);
-  const donorIds = beneficiaries.reduce<string[]>((acc, b) => [...acc, ...b.donors], []);
+  const beneficiaries = await findEligibleBeneficiaries(db, periodLimit, periodLength, filter);
+  let donorIds;
+  if (beneficiaries.length && beneficiaries[0].donors) {
```

Comment:
I think you should add an argument to this function that states whether or not this should use all donors, and then use that on the if-condition. Or maybe the be whether or not this distribution is vetted

```js
const filter = onlyVettedBeneficiaries ? { isVetted: true, beneficiaryStatus: 'verified' } : {};
const beneficiaries = await findEligibleBeneficiaries(db, periodLimit, periodLength, filter);
if (onlyVettedBeneficiaries) {
  const donors = await db.collection(USERS_COLL).find({ roles: [ $in: ['donor'] })...
}
else {
   donorIds = beneficiaries.reduce((acc, b) => ...));
}

Reply:
Also, you should ideally exclude donors who have transaction blocks (but this is probably already handled in `computeDonorBalances`)